### PR TITLE
Add collation info for ANALYTICS_DB on primary key violations

### PR DESCRIPTION
**Pull request by RakhithaRR**
### Purpose
This PR adds information regarding adding collation level case sensitive support for databases in MySQL and MSSQL.

Related issue: https://github.com/wso2/streaming-integrator/issues/158
https://github.com/wso2/docs-apim/issues/2447

### CODE REVIEW

**Review by ruks — APPROVED** [review]

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-310.md [hunk 1]
```diff
@@ -1877,6 +1877,17 @@ Follow the instructions below to configure WSO2 API Manager for the WSO2 API-M A
     !!! note
         When performing Analytics migration in WSO2 API-M Analytics 3.1.0, you need to set the **defaultAutoCommit** value to **true** in the `APIM_ANALYTICS_DB` and `WSO2AM_STATS_DB` datasource configurations.
 
+    !!! info "Support case sensitive primary keys"
```
[the remark this answers is not in the document]
Comment (RakhithaRR) [reply en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-310.md]: Fixed with 4f1ad5a

**Comment by Mariangela** [comment]
@ruks , can I have more clarification on this PR.

- I checked the comments and it looks like based on your review comment Rakhitha has changed the PR from what is said in the [Git issue](https://github.com/wso2/docs-apim/issues/2447) if the change correct or has he misunderstood your comment.

- This statement is very vague
"Sometimes due to the primary key length exceeding, you have to use different encoding types that take lower bytes (than UTF8)."
Q1: what is meant by "Sometimes"
Q2: Does the primary key length only exceed for the MySQL and MS SQL DB
Q3: can we change [1] to [2]?
[1] Sometimes due to the primary key length exceeding, you have to use different encoding types that take lower bytes (than UTF8).

[2] When working with MySQL and MS SQL the primary key length exceeds the stipulated maximum length. Therefor when working with  MySQL and MS SQL DBs  you have to use different encoding types that take lower bytes (than UTF8).

**Comment by ruks** [comment]
@Mariangela, it depends on the DB. Based on the DB version, type, and configuration you may face this exception. Also depends on the analytics data also you may notice this exception. I think that's why we used the term sometimes. 
And this is not limited to MySQL and MSSQL. These are the configurations we found for these DB types. So we are mentioning them as a reference here.

**Comment by Mariangela** [comment]
@RakhithaRR , is this fix applicable to 3.0.0 and 3.2.0? If yes, please send PRs.

**Comment by RakhithaRR** [comment]
@Mariangela I'll send the PRs for 3.2.0 and master

**Pull request merged**